### ISSUE TITLE
Custom columns of tag type should be treated like tags in Advanced Search

### ISSUE THREAD

**Issue by arieol** [janeczku/calibre-web#1626]
**Is your feature request related to a problem? Please describe.**

In advanced search, you can filter the regular "tags" using two lists of checkboxes, to include or exclude them. But custom columns that should be treated like tags use a textbox (for columns with unlimited possible values) or a dropdown list (for columns with limited possible values). And thus, you cannot exclude a value from a custom column.

Examples:

1) Since filtering is always done using "contains" and not "equals" method, then if I have a custom column "_publication type_" of type "like tags with limited set of values" that contains values "_novel_" and "_novella_", then when I search for "_novel_", the books from both of those categories are returned (since: "_noval{la}_").

2) Similarly, if I have a custom column "_warnings_" of tyle "like tags", then I cannot exclude books that contain both "_love triangle_" and "_instalove_".

3) You cannot choose more than one value from columns that are of the type "list of tags": if I have a custom column of type "_content_", I cannot (easily) search for the books that have both "_vampires_" and "_witches_" but don't contain "_werewolves_".

**Describe the solution you'd like**

Custom columns of "tag" type should be treated exactly like regular tags in Advanced Search. It would look neater and be easier to use (for example, choosing values from a dropdown requires one additional "click"). Also, when it comes to custom "tag" columns without "limited set of possible values", the user may not have any idea what to type (what terminology and definitions the library's admin is choosing to use), because there is only a textbox there (unlike a list of checkboxes as with the "regular" tags).

I think it would be best if you could choose how "custom tag columns" are to be treated in library's settings (if textboxes or checkboxes are to be used in advanced search). This way, people who have custom columns with long lists of values or with long entries could disable this functionality, so it doesn't clutter the GUI.

**Comment by kctdfh** [comment]
I agree that this is essential for people who have heavily customized Calibre libraries. Paging @OzzieIsaacs to see if this is a low hanging fruit or not. I've seen it requested often and as recently as a couple of months ago (#2459) but it's never been addressed by any of the maintainers! Thanks for having a look!
